src/Microsoft.DotNet.Build.Tasks/BinClashLogger.cs

```diff
@@ -0,0 +1,231 @@
+                    }
+                }
+            }
+        }
+
+        private void ApplyParameter(string name, string value)
+        {
+            switch(name.ToLower())
+            {
+                case "logfile":
+                    _logFile = value;
+                    break;
+                case "excptiononerror":
+                    _exceptionOnError = Boolean.Parse(value);
+                    break;
+                case "outputtostderr":
+                    _outputToStdErr = Boolean.Parse(value);
+                    break;
+                case "ignorenonexistenttargetpaths":
+                    _ignoreNonExistentTargetPaths = Boolean.Parse(value);
+                    break;
+                default:
+                    // ignore unrecognized parameters
+                    break;
+            }
+        }
+
+        private void ProjectStarted(object sender, ProjectStartedEventArgs e)
+        {
+            var state = new ProjectState(e);
+            _projectHistory.Add(e.BuildEventContext.ProjectContextId, state);
+        }
+
+        private void TargetStarted(object sender, TargetStartedEventArgs e)
+        {
+            // we only care about the build target
+            if (e.TargetName == "Build")
+            {
+                var state = _projectHistory[e.BuildEventContext.ProjectContextId];
+                state.RanBuild = true;
+            }
+        }
+
+        public void Shutdown()
+        {
+            int clashes = 0;
+            Dictionary<string, ProjectState> clashMap = new Dictionary<string, ProjectState>();
+            foreach (var state in _projectHistory.Values.Where(s => s.RanBuild && !String.IsNullOrEmpty(s.TargetPath)))
+            {
+                if (_ignoreNonExistentTargetPaths && !File.Exists(state.TargetPath))
+                {
+                    continue;
+                }
+
+                ProjectState clashingProject = null;
+                if (!clashMap.TryGetValue(state.TargetPath, out clashingProject))
+                {
+                    clashMap[state.TargetPath] = state;
+                }
+                else
+                {
+                    StringBuilder errorMessage = new StringBuilder($"Error : Multiple projects built twice with the same target path {state.TargetPath}.");
+                    errorMessage.AppendLine();
+                    errorMessage.AppendLine(GetProjectStack(state));
+                    errorMessage.AppendLine(GetProjectStack(clashingProject));
+                    errorMessage.AppendLine();
+                    WriteError(errorMessage.ToString());
+                    clashes++;
+                }
+            }
+
+            if (_fileWriter != null)
+            {
+                _fileWriter.Dispose();
+            }
+
+            if (_exceptionOnError && (clashes != 0))
+            {
+                throw new Exception($"{clashes} bin clashes were detected during the build.");
```

Comment:
I spent quite a lot of time trying to find a way to gracefully error-out but I couldn't come up with anything.  I tried forwardinglogger/nodelogger which can let me generate new events but those events only flow to my central logger, not the default loggers.  I could have created some task that runs at the end of the build and communicate with it, but that felt less general purpose.